src/NzbDrone.Common/Http/HttpRequest.cs

```diff
@@ -76,5 +77,15 @@ public void SetContent(string data)
             var encoding = HttpHeader.GetEncodingFromContentType(Headers.ContentType);
             ContentData = encoding.GetBytes(data);
         }
+
+        public void SetNetworkCredential(NetworkCredential networkCredential)
+        {
+            if (networkCredential != null)
```

Comment:
This should never be called if network credential is null, if it is, it should explode.